aws/resource_aws_cognito_user_pool_client.go

```diff
@@ -0,0 +1,215 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCognitoUserPoolClient() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoUserPoolClientCreate,
+		Read:   resourceAwsCognitoUserPoolClientRead,
+		Update: resourceAwsCognitoUserPoolClientUpdate,
+		Delete: resourceAwsCognitoUserPoolClientDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"generate_secret": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"user_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"explicit_auth_flows": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
```

Comment:
It looks like this field is updatable (at least it's present in `UpdateUserPoolClientInput`) - is there any reason for making it ForceNew?

Reply:
Not sure which field this refers to?

Reply:
Ah, sorry - I missed that - so in that case it's practically correct, but it's `false` by default so we can just omit it.